cfilament/src/API.h

```diff
@@ -0,0 +1,162 @@
+// For pointer types and enums this should not affect the ABI
+#include <math/vec3.h>
+#include <math/mat4.h>
+#include <filament/Color.h>
+#include <filament/Box.h>
+
+using FMat4 = math::mat4;
+using FMat4f = math::mat4f;
+using FFloat3 = math::float3;
+using FFloat4 = math::float4;
+using FBox = filament::Box;
+using FLinearColor = filament::LinearColor;
+using FLinearColorA = filament::LinearColorA;
+
+// We cannot use forward declarations for nested enums and classes, sadly.
+#include <filament/Camera.h>
+#include <filament/Fence.h>
+#include <filament/MaterialEnums.h>
+#include <filament/IndexBuffer.h>
+#include <filament/LightManager.h>
+#include <filament/IndirectLight.h>
+#include <filament/RenderableManager.h>
+#include <filament/Skybox.h>
+#include <filament/VertexBuffer.h>
+#include <filament/View.h>
+#include <filament/Stream.h>
+#include <filament/Texture.h>
+
+using FIndexBufferBuilder = filament::IndexBuffer::Builder;
+using FLightManagerBuilder = filament::LightManager::Builder;
+using FIndirectLightBuilder = filament::IndirectLight::Builder;
+using FRenderableManagerBuilder = filament::RenderableManager::Builder;
+using FRenderableManagerBone = filament::RenderableManager::Bone;
+using FVertexBufferBuilder = filament::VertexBuffer::Builder;
+using FSkyboxBuilder = filament::Skybox::Builder;
+using FStreamBuilder = filament::Stream::Builder;
+using FTextureBuilder = filament::Texture::Builder;
+
+using FCamera = filament::Camera;
+using FCameraProjection = filament::Camera::Projection;
+using FCameraFov = filament::Camera::Fov;
+using FFence = filament::Fence;
+using FFenceType = filament::Fence::Type;
+using FFenceStatus = filament::Fence::FenceStatus;
+using FFenceMode = filament::Fence::Mode;
+
+using FMaterialShading = filament::Shading;
+using FMaterialInterpolation = filament::Interpolation;
+using FMaterialBlendingMode = filament::BlendingMode;
+using FMaterialVertexDomain = filament::VertexDomain;
+using FMaterialCullingMode = filament::driver::CullingMode;
+using FMaterialParameterType = filament::driver::UniformType;
+using FMaterialSamplerType = filament::driver::SamplerType;
+using FMaterialPrecision = filament::driver::Precision;
+using FPixelDataFormat = filament::driver::PixelDataFormat;
+using FPixelDataType = filament::driver::PixelDataType;
+using FTextureFormat = filament::driver::TextureFormat;
+using FSamplerType = filament::driver::SamplerType;
+using FTextureUsage = filament::driver::TextureUsage;
+using FTextureCubemapFace = filament::driver::TextureCubemapFace;
+
+using FIndexBufferIndexType = filament::IndexBuffer::IndexType;
+
+using FLightManagerType = filament::LightManager::Type;
+
+using FPrimitiveType = filament::driver::PrimitiveType;
+using FElementType = filament::driver::ElementType;
+using FVertexAttribute = filament::VertexAttribute;
+using FAntiAliasing = filament::View::AntiAliasing;
+using FDepthPrepass = filament::View::DepthPrepass;
+
+namespace filament {
+  class Engine;
```

Comment:
Code style rules: we use 4 spaces for normal indentation and 8 for continuation indentation.